### PR TITLE
Restructure Train by Voice composer into a 3-step accordion

### LINKED ISSUE #648: [BUG] train by voice and teach pronunciation not working
### Describe the bug

The trained by voice and teach pronunciation features under custom dictionary seem to be unavailable or not working. Have checked discussion forums for open threads. None exist 

### Reproduction steps

1. Select custom dictionary 
2. Select train by voice: No prompts in the interface or elsewhere to do anything
3. Select basic/advanced (research preview): Start button's state remains inactive 

### Expected behavior

I expected a prompt in the interface 

### Actual behavior

No interactive elements 

### App Version

1.6.4

### macOS Version

macOS Tahoe 26.5.1

### Architecture

Apple Silicon

### Logs or crash report

```text

```

### Screenshots or screen recording

[CleanShot 2026-07-18 at 04.54.13.png](https://github.com/user-attachments/assets/660b5414-28f5-4326-b63a-23e598ae609b)

### PR DESCRIPTION
**Pull request by shreeraman96**
## Description
Fixes #648, where **Train by Voice / Teach Pronunciation** under Custom Dictionary showed no interactive prompts and left the Start button permanently inactive. This replaces the flat replacement composer with a guided 3-step accordion (① Word → ② Record → ③ Verify & Save) that always surfaces the next action, and hardens the underlying step state machine.

Changes:
- Restructure the composer into a 3-step accordion with per-step status glyphs; extract pure step-derivation + readiness logic into a new SwiftUI-free, unit-tested `DictionaryTrainingStepModel` (`DictionaryTrainingSnapshot` groups the training state)
- Reset the `hasReachedVerify` latch on every teardown path (Try Again, capture removal, mode toggle) so Verify never dead-ends with Save disabled
- Reset `lastAnnouncedTrainingStep` on teardown and announce any forward step edge (incl. `word → verify`) for correct VoiceOver cues
- Make Record/Verify headers non-interactive while the word is empty, and keep `Verify & save` locked until the derived step actually reaches verify so it can't be opened with nothing recorded and Add Replacement disabled; surface save-failure errors in the Verify step
- Simplify the accordion copy: step titles carry the instruction (`Type the word to teach` / `Say it 3 times` / `Verify & save`), per-step subtitles and the word-echoing readiness strings are gone
- Move the `trainingReplacement` `onChange` onto the always-mounted accordion so programmatic word writes still reset progress
- Add unit tests for step derivation, the readiness predicates, and edge cases (empty-word latch, case-insensitivity, pronunciation boundaries)

## Type of Change
- [x] 🐞 Bug fix
- [ ] ✨ New feature
- [ ] 💥 Breaking change
- [ ] 🧹 Chore
- [ ] 📝 Documentation update

## Related Issue or Discussion
Closes #648

## Testing
- [ ] Tested on Intel Mac
- [x] Tested on Apple Silicon Mac
- [x] Tested on macOS version: 26.5 (Tahoe)
- [x] Ran linter locally: `swiftlint --strict --config .swiftlint.yml Sources` — 0 violations
- [x] Ran formatter locally: `swiftformat --config .swiftformat` — changed files pass
- [x] Ran tests locally: `DictionaryTrainingStepModelTests` — 25/25 passing

#### Step 1 : Enter the word to teach
image

#### Step 2 : Record 
image

#### Step 3 : Verify & save
image

#### Manual Add

image

#### Research preview 
image

#### Before all shown at once 

image

## Notes
- No shared-package (FluidIntelligence) changes — no cross-repo impact.
- Findings from a two-model deep review + PR bot review (latch not reset on teardown, missing Verify-step error feedback, VoiceOver announcement gaps, empty-word dead-ends) are all fixed.

### CODE REVIEW

File: Sources/Fluid/UI/CustomDictionaryView.swift [hunk 1]
```diff
@@ -2536,13 +2812,43 @@ private enum DictionaryTrainingCopy {
...
             : "Keep trying until FluidVoice gets \(target) right 3 times in a row."
     }
+
+    // MARK: - Train by Voice accordion
+
+    static func stepTitle(_ step: DictionaryTrainingStep) -> String {
+        switch step {
+        case .word: return "Word"
+        case .record: return "Record"
+        case .verify: return "Verify & Save"
```
Comment (grohith327) [inline Sources/Fluid/UI/CustomDictionaryView.swift]: We can remove the subtitle and keep this more detailed but concise so that the user knows what is needed in each step

File: Sources/Fluid/UI/CustomDictionaryView.swift [hunk 2]
```diff
@@ -2285,13 +2211,361 @@ private extension CustomDictionaryView {
...
+            (self.trainingSampleCount > 0 || !self.trainingPronunciationEnrollments.isEmpty)
+    }
+
+    /// A step header is tappable unless the recording lock pins `.record`, or the
+    /// word is still empty (Record/Verify have nothing to act on and would strand
+    /// the user on a disabled panel with no caption).
+    func isTrainingStepInteractive(_ step: DictionaryTrainingStep) -> Bool {
+        if self.isTrainingRecordingLocked && step != .record { return false }
+        if step != .word && self.normalizedTrainingReplacement.isEmpty { return false }
+        return true
```
Comment (grohith327) [inline Sources/Fluid/UI/CustomDictionaryView.swift]: This allows me to "verify and save" step (although I cant add replacement) even if I haven't recorded anything

**Review by grohith327 — COMMENTED** [review]
Had a quick glance, added few comments. A few more requests for changes:

When someone types a word, it shows up in many places, not needed since they know what they are typing

Image

The steps in record tab are too detailed, I would recommend shrinking it a bit to be specific 

Image

These are probably not entirely related to your changes but would recommend doing the clean up in one go

File: Sources/Fluid/UI/CustomDictionaryView.swift (hunk 2 above)
Comment (shreeraman96) [reply Sources/Fluid/UI/CustomDictionaryView.swift]: Fixed in 32aa246.

The tap predicate moved out of the view into `DictionaryTrainingStepModel.isStepInteractive`, and `.verify` is now gated on the derived step actually reaching verify:

```swift
if isRecordingLocked { return step == .record }
if step != .word, wordIsEmpty { return false }
if step == .verify, derived != .verify { return false }
return true
```

`derivedStep` only returns `.verify` when `ready || alreadyCorrect || hasReachedVerify`, all of which require 3 covered captures or 3 pronunciation enrollments — so with nothing recorded the header is inert. The `hasReachedVerify` latch keeps it tappable after a post-ready missed capture so Try Again stays reachable.

Covered by `testVerifyHeaderIsNotTappableBeforeAnythingIsRecorded`, `testVerifyHeaderIsTappableOnceDerivedStepReachesVerify`, `testRecordAndVerifyHeadersAreLockedWhileWordIsEmpty`, `testRecordingLockLeavesOnlyRecordTappable`. Full suite: 25/25 passing.

File: Sources/Fluid/UI/CustomDictionaryView.swift (hunk 1 above)
Comment (shreeraman96) [reply Sources/Fluid/UI/CustomDictionaryView.swift]: Done in 32aa246 — subtitles are gone entirely.

The instruction now lives in the step title itself, so each header states what is needed without a second line:

- ① `Type the word to teach`
- ② `Say it 3 times`
- ③ `Verify & save`

`DictionaryTrainingStepCopy` (the subtitle generator) and `wordStepSubtitle` were deleted along with `accessibilityValue`. The in-panel instructions shrank from 4 steps to 2 (`Press Start, say the word naturally, then pause.` / `Repeat until the ring reaches 3/3.`), and they drop out once training is ready or already correct — the readiness ring and its caption carry the outcome from there. Readiness copy also no longer echoes the typed word back, since it is already on screen in step ① and the final-output panel.

File: Sources/Fluid/UI/CustomDictionaryView.swift [hunk 3]
```diff
@@ -481,27 +544,177 @@ struct CustomDictionaryView: View {
...
+        return nil
+    }
+
+    @ViewBuilder
+    private var trainingVerifyStepBody: some View {
+        VStack(alignment: .leading, spacing: self.theme.metrics.spacing.sm) {
+            self.trainingFinalOutputPanel
 
             Button {
                 Task { await self.addTrainedReplacement() }
```
[the remark this answers is not in the document]
Comment (shreeraman96) [reply Sources/Fluid/UI/CustomDictionaryView.swift]: Addressed in d7381c4, though not by duplicating the footer into the Verify body.

The error label is hoisted out of the collapsible step bodies and rendered once in the accordion container (`CustomDictionaryView.swift:2303-2310`), directly after the three headers:

```swift
// Always-visible error label: rendered once outside the collapsible step
// bodies so a recording/processing failure is surfaced regardless of
// which step is expanded.
if self.trainingHasError {
    Label(self.trainingStatusMessage, systemImage: "exclamationmark.triangle.fill")
        .font(self.theme.typography.caption)
        .foregroundStyle(self.theme.palette.warning)
}
```

So a failed `addTrainedReplacement()` with advanced voice matching enabled surfaces `trainingStatusMessage` while the accordion sits on Verify, and the same holds for record/processing failures on any expanded step — one rendering site instead of per-step copies.
